### PR TITLE
Implement document configuration loading of forms

### PR DESCRIPTION
**Pull request by cozej4**

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

File: opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java [hunk 1]
```diff
@@ -5,10 +5,13 @@
 
 import org.jetbrains.annotations.NotNull;
 import org.smartregister.chw.core.utils.ChildDBConstants;
+import org.smartregister.chw.core.utils.CoreConstants;
 import org.smartregister.chw.referral.model.BaseReferralRegisterFragmentModel;
+import org.smartregister.chw.util.ChwDBConstants;
 import org.smartregister.chw.util.Constants;
 import org.smartregister.cursoradapter.SmartRegisterQueryBuilder;
 import org.smartregister.family.util.DBConstants;
+import org.smartregister.repository.TaskRepository;
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java:14]: Codacy Issue found: [Avoid unused imports such as 'org.smartregister.repository.TaskRepository'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601)

File: opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java [hunk 2]
```diff
@@ -5,10 +5,13 @@
 
 import org.jetbrains.annotations.NotNull;
 import org.smartregister.chw.core.utils.ChildDBConstants;
+import org.smartregister.chw.core.utils.CoreConstants;
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java:8]: Codacy Issue found: [Avoid unused imports such as 'org.smartregister.chw.core.utils.CoreConstants'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

File: opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java [hunk 3]
```diff
@@ -52,7 +54,7 @@ public String mainSelect(@NonNull String tableName, @NonNull String mainConditio
         columnList.add("T2." + DBConstants.KEY.FIRST_NAME + " || " + "' '" + " || " + "T2." + DBConstants.KEY.MIDDLE_NAME + " || " + "' '" + " || " + "T2." + DBConstants.KEY.LAST_NAME + " AS " + DBConstants.KEY.FAMILY_HEAD);
         columnList.add(Tables.REFERRAL + "." + Key.REFERRAL_SERVICE + "  AS " + Key.REFERRAL_SERVICE);
         columnList.add(Tables.REFERRAL + "." + Key.REFERRAL_DATE);
-        columnList.add(Tables.REFERRAL + "." + Key.REFERRAL_STATUS);
+        columnList.add(Constants.TABLE_NAME.TASK + "." + org.smartregister.chw.core.utils.ChwDBConstants.TaskTable.BUSINESS_STATUS + " AS " + Key.REFERRAL_STATUS);
```
Comment (ellykits) [inline opensrp-chw/src/main/java/org/smartregister/chw/model/ReferralRegisterFragmentModel.java:57]: Using the qualified name is not necessary at this point.

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/interactor/LoginJobSchedulerProvider.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/job/ChwJobCreator.java  1
         

Complexity decreasing per file
==============================
+ opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyRegisterActivity.java  -2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FamilyPlanningMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/PncMemberProfileActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberProfileActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5439601&bid=17838590)

**Pull request merged**